src/handlers/motions/motionFinalized/helpers.ts

```diff
@@ -96,3 +116,84 @@ export const linkPendingDomainMetadataWithDomain = async (action: string, colony
     });
   }
 };
+
+export const linkPendingColonyMetadataWithColony = async (
+  pendingColonyMetadata: ColonyMetadata,
+  colonyAddress: string,
+): Promise<void> => {
+  const currentColonyMetadata = await query<ColonyMetadata>(
+    'getColonyMetadata',
+    {
+      id: colonyAddress,
+    },
+  );
+
+  if (!currentColonyMetadata) {
+    console.error(
+      `Could not find the current metadata for the colony: ${colonyAddress}. This is a bug and should be investigated.`,
+    );
+    return;
+  }
+
+  const {
+    haveTokensChanged,
+    hasAvatarChanged,
+    hasWhitelistChanged,
+    newDisplayName,
+    oldDisplayName,
+  } = pendingColonyMetadata.changelog?.[0] ?? {};
+
+  const updatedMetadata = {
+    ...currentColonyMetadata,
+  };
+
+  /*
+   * Here, we update metadata as granularly as possible so that we don't overwrite state changes that occured
+   * after this motion was created.
+   */
```

Comment:
As the comment suggests, this should fix the problem of one motion overwriting the state changes of another motion that was created after it, but finalized before it is finalized. To be tested in #433

Reply:
If this works as I imagine, we can refactor how we store Domain metadata to follow the same pattern.

Reply:
https://github.com/JoinColony/block-ingestor/issues/76